test/replication-luatest/gh_8931_box_info_election_leader_name_test.lua

```diff
@@ -0,0 +1,89 @@
+local t = require('luatest')
+local cluster = require('luatest.replica_set')
+local server = require('luatest.server')
+
+local g = t.group('gh-8931')
+--
+-- gh-8931:
+-- There is the leader field in the given informational API, but it is easier
+-- to a human to understand human-readable names. We have instance names
+-- since #5029. Let's show them in this API.
+--
+g.before_all(function(cg)
+    cg.cluster = cluster:new({})
+    local cfg = {
+        replication = {
+            server.build_listen_uri('node1', cg.cluster.id),
+            server.build_listen_uri('node2', cg.cluster.id),
+        },
+        instance_name = 'node1'
+    }
+    cg.node1 = cg.cluster:build_and_add_server({alias = 'node1', box_cfg = cfg})
+    cfg.instance_name = 'node2'
+    cg.node2 = cg.cluster:build_and_add_server({alias = 'node2', box_cfg = cfg})
+    cg.cluster:start()
+end)
+
+g.before_test('test_election_mode_candidate', function(cg)
+    cg.node1:exec(function()
+        box.cfg{
+            election_mode = 'candidate',
+            instance_name = 'node1',
+        }
+    end)
+    cg.node2:exec(function()
+        box.cfg{
+            election_mode = 'candidate',
+            instance_name = 'node2',
+        }
+    end)
+    cg.node1:wait_until_election_leader_found()
+    cg.node2:wait_until_election_leader_found()
+end)
+
+g.test_election_mode_candidate = function(cg)
+    local leader = cg.cluster:get_leader()
+    t.assert(leader ~= nil, 'Cluster elected a leader')
+    t.assert_equals(box.info.election.leader_name, leader.name)
+end
+
+g.before_test('test_election_mode_off', function(cg)
+    cg.node1:exec(function()
+        box.cfg{election_mode = 'off',}
+    end)
+    cg.node2:exec(function()
+        box.cfg{election_mode = 'off',}
+    end)
+    cg.node1:wait_for_election_state('follower')
+    cg.node2:wait_for_election_state('follower')
+end)
+
+g.test_election_mode_off = function(cg)
+    t.assert(cg.cluster:get_leader() == nil, 'Cluster has no leader')
+    t.assert_equals(box.info.election.leader_name, nil)
+end
+
+g.before_test('test_leader_has_no_name', function(cg)
+    cg.node1:exec(function()
+        box.cfg{
+            election_mode = 'candidate',
+            instance_name = box.NULL,
```

Comment:
Setting instance name to `box.NULL` might not do what you expect. It simply means that the config doesn't force any particular instance name, but the existing instance name, if set, remains.

I suggest you first test the case without instance names, and then set the instance names. Or you may even create two separate clusters with separate configuration for the tests.

Reply:
No way to delete an instance name on the go?

Reply:
It's possible but quite hard to do. Names are not supposed to be dropped during normal operation. The option is left only for incident recovery.